test_spack.py

```diff
@@ -0,0 +1,214 @@
+    'dawn' : {},
+    'dawn4py' : {},
+    'dusk' : {},
+    'dyicon_benchmarks' : {},
+    'ecbuild' : {},
+    'eccodes' : {},
+    'eckit' : {},
+    'gridtools' : {},
+    'icon' : {
+        # 'git clone --recursive git@gitlab.dkrz.de:icon/icon-cscs.git \
+        #     && cd icon-cscs \
+        #     && mkdir pgi_cpu \
+        #     && cd pgi_cpu \
+        #     && touch a_fake_file.f90 \
+        #     && spack dev-build -u build icon@dev-build%pgi config_dir=./.. icon_target=cpu', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+        # 'git clone --recursive git@gitlab.dkrz.de:icon/icon-cscs.git \
+        #     && cd icon-cscs \
+        #     && mkdir pgi_gpu \
+        #     && cd pgi_gpu \
+        #     && touch a_fake_file.f90 \
+        #     && spack dev-build -u build icon@dev-build%pgi config_dir=./.. icon_target=gpu', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+    },
+    'icondusk-e2e' : {},
+    'icontools' : {},
+    'int2lm' : {
+        'spack install int2lm@c2sm_master%pgi', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+        'spack install int2lm@org_master%pgi pollen=False', # Listed in https://c2sm.github.io/spack-c2sm/QuickStart.html
+    },
+    'libgrib1' : {},
+    'mpich' : {},
+    'oasis' : {},
+    'omnicompiler' : {},
+    'omni-xmod-pool' : {},
+    'openmpi' : {},
+    'serialbox' : {},
+    'xcodeml-tools' : {},
+    'zlib_ng' : {},
+}
+
+def UseCases(packages : set):
+    return {case for p in packages for case in use_cases[p]}
+
+# Maps packages to Set[packages it depends on].
+dependencies = {
+    'atlas' : {'ecbuild', 'eckit'},
+    'atlas_utilities' : {'atlas', 'eckit'},
+    'claw' : {},
+    'cosmo' : {'cuda', 'serialbox', 'libgrib1', 'cosmo-grib-api-definitions', 'cosmo-eccodes-definitions', 'omni-xmod-pool', 'claw', 'zlib_ng'},
+    'cosmo-dycore' : {'gridtools', 'serialbox', 'cuda'},
+    'cosmo-eccodes-definitions' : {'eccodes'},
+    'cosmo-grib-api' : {},
+    'cosmo-grib-api-definitions' : {'cosmo-grib-api'},
+    'cuda' : {},
+    'dawn' : {},
+    'dawn4py' : {},
+    'dusk' : {'dawn4py'},
+    'dyicon_benchmarks' : {'atlas_utilities', 'atlas', 'cuda'},
+    'ecbuild' : {},
+    'eccodes' : {},
+    'eckit' : {'ecbuild'},
+    'gridtools' : {'cuda'},
+    'icon' : {'serialbox', 'eccodes', 'claw'},
+    'icondusk-e2e' : {'atlas_utilities', 'dawn4py', 'dusk', 'atlas', 'cuda'},
+    'icontools' : {'eccodes', 'cosmo-grib-api'},
+    'int2lm' : {'cosmo-grib-api-definitions', 'cosmo-eccodes-definitions', 'libgrib1'},
+    'libgrib1' : {},
+    'log' : {},
+    'mpich' : {},
+    'oasis' : {},
+    'omnicompiler' : {},
+    'omni-xmod-pool' : {},
+    'openmpi' : {},
+    'serialbox' : {},
+    'xcodeml-tools' : {},
+    'zlib_ng' : {},
+}
+
+def Dependents(packages : set):
+    return {parent for parent, children in dependencies.items() if not packages.isdisjoint(children)}
+
+# Maps commands to Set[packages]
+commands_to_packages = {
```

Comment:
there is some a repetition. How about: 
```
all=['atlas', 'atlas_utilities', 'claw', 'cosmo', 'cosmo-dycore', 'cosmo-eccodes-definitions',
        'cosmo-grib-api', 'cosmo-grib-api-definitions', 'cuda', 'dawn', 'dawn4py', 'dusk',
        'dyicon_benchmarks', 'ecbuild', 'eccodes', 'eckit', 'gridtools', 'icon', 'icondusk-e2e',
        'icontools', 'int2lm', 'libgrib1', 'mpich', 'oasis', 'omnicompiler', 'omni-xmod-pool',
        'openmpi', 'serialbox', 'xcodeml-tools', 'zlib_ng']
commands_to_packages = {p:p for p in all}
commands_to_packages.update({'all': all})
``

Reply:
I like it. I'll include it in the next commit.